x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/get_datastream_settings/index.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { datasetQualityPrivileges, dataStreamService } from '../../../services';
+import { DataStreamSettings } from '../../../../common/api_types';
+import { getDataStreamCreatedOn } from './get_datastream_created_on';
+
+export async function getDataStreamSettings({
+  esClient,
+  dataStream,
+}: {
+  esClient: ElasticsearchClient;
+  dataStream: string;
+}): Promise<DataStreamSettings> {
+  const [createdOn, [dataStreamInfo], datasetUserPrivileges] = await Promise.all([
+    getDataStreamCreatedOn(esClient, dataStream),
+    dataStreamService.getMatchingDataStreams(esClient, dataStream),
+    datasetQualityPrivileges.getDatasetPrivileges(esClient, dataStream),
+  ]);
+
+  const integration = dataStreamInfo?._meta?.package?.name;
+  const lastBackingIndex = dataStreamInfo?.indices?.slice(-1)[0];
```

Comment:
> [!TIP]
> Seems like `.pop` it's what we need here.
```ts
const lastBackingIndex = dataStreamInfo?.indices?.pop();
```

Reply:
But `pop()` would mutate the array, which might have unintended consequences. I'd rather stay with a side-effect-free operation.

Reply:
It should be safe since the life span of the `dataStreamInfo` is only scoped to this function, but I get your concern and make sense in case we'll extend the capabilities of this function.

P.S. also `dataStreamInfo.indices.at(-1)` is a valid option as it only reference the last element not mutating the list.